[WIP] [Enhancement] Enable memtable sync write when there are too many memtables in the queue

xiaoyong-z opened the pull request:
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #6735

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->
When the concurrent stream load is fast enough, memtable write and flush thread can not catch up, continue to write the memtable will only make the memtable queue even longer, resulting in a large number of memtable in the memory.
In order to solve this problem, we can transfer from the async memtable table write to sync memtable write. Let the import speed to be slowed down and flush the memtable table in the queue.

imay commented:
I have a question. After your change, the write operation will be write sync. The RPC thread will be held all the time for this RPC call. If all the RPC thread is held by this sync write, then other RPC call can not get thread to execute. Then this backend will hang all the time.
So I think this is not a good idea to do sync write in the RPC thread, It will be better to tell the sender send it later and slower.

xiaoyong-z committed:
[Enhancement] Enable memtable sync write when there are too many memtables in the queue

When the concurrent stream load is fast enough, memtable write and flush thread can not catch up, continue to write the memtable will only make the memtable queue even longer, resulting in a large number of memtable in the memory.
In order to solve this problem, we can transfer from the async memtable table write to sync memtable write. Let the import speed to be slowed down and flush the memtable table in the queue.

Signed-off-by: xyz <a997647204@gmail.com>

chaoyli commented on be/src/storage/async_delta_writer.cpp:
```diff
@@ -83,11 +100,47 @@ void AsyncDeltaWriter::write(const AsyncDeltaWriterRequest& req, AsyncDeltaWrite
+    StarRocksMetrics::instance()->memtable_execution_queue_count.increment(1);
+}
+
+void AsyncDeltaWriter::sync_write(const AsyncDeltaWriterRequest& req, AsyncDeltaWriterCallback* cb) {
```
I think it's better to use a request/response to limit bandwidth to send data.
It's better to consider the disk I/O performance to throttle.
